clients/redigo.go

```diff
@@ -0,0 +1,389 @@
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONNumMultBy to increment a number by provided amount
+//
+// ReJSON syntax:
+// 	JSON.NUMMULTBY <key> <path> <number>
+//
+func (r *Redigo) JSONNumMultBy(key, path string, number int) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_NUMMULTBY, key, path, number)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONStrAppend to append a jsonstring to an existing member
+//
+// ReJSON syntax:
+// 	JSON.STRAPPEND <key> [path] <json-string>
+//
+func (r *Redigo) JSONStrAppend(key, path, jsonstring string) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_STRAPPEND, key, path, jsonstring)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONStrLen to return the length of a string member
+//
+// ReJSON syntax:
+// 	JSON.STRLEN <key> [path]
+//
+func (r *Redigo) JSONStrLen(key, path string) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_STRLEN, key, path)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONArrAppend to append json value into array at path
+//
+// ReJSON syntax:
+// 	JSON.ARRAPPEND <key> <path> <json> [json ...]
+//
+func (r *Redigo) JSONArrAppend(key, path string, values ...interface{}) (res interface{}, err error) {
+
+	if len(values) == 0 {
+		return nil, rjs.NeedAtleastOneArg
+	}
+	args := make([]interface{}, 0)
+	args = append(args, key, path)
+	args = append(args, values...)
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_ARRAPPEND, args...)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONArrLen returns the length of the json array at path
+//
+// ReJSON syntax:
+// 	JSON.ARRLEN <key> [path]
+//
+func (r *Redigo) JSONArrLen(key, path string) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_ARRLEN, key, path)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONArrPop removes and returns element from the index in the array
+// to pop last element use rejson.PopArrLast
+//
+// ReJSON syntax:
+// 	JSON.ARRPOP <key> [path [index]]
+//
+func (r *Redigo) JSONArrPop(key, path string, index int) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_ARRPOP, key, path, index)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONArrIndex returns the index of the json element provided and return -1 if element is not present
+//
+// ReJSON syntax:
+// 	JSON.ARRINDEX <key> <path> <json-scalar> [start [stop]]
+//
+func (r *Redigo) JSONArrIndex(key, path string, jsonValue interface{}, optionalRange ...int) (
+	res interface{}, err error) {
+
+	args := []interface{}{key, path, jsonValue}
+
+	ln := len(optionalRange)
+	switch {
```

Comment:
Do we need this switch statement ? You could just do `args = append(args, optionalRange...)`.

Reply:
Yes, because the function, ArrIndex can only accept at most 2 optional fields.